cli/azd/pkg/azapi/stack_deployments.go

```diff
@@ -242,42 +244,11 @@ func (d *StackDeployments) DeployToSubscription(
 	tags map[string]*string,
 	options map[string]any,
 ) (*ResourceDeployment, error) {
-	client, err := d.createClient(ctx, subscriptionId)
-	if err != nil {
-		return nil, err
-	}
-
-	templateHash, err := d.CalculateTemplateHash(ctx, subscriptionId, armTemplate)
-	if err != nil {
-		return nil, fmt.Errorf("failed to calculate template hash: %w", err)
-	}
-
-	clonedTags := maps.Clone(tags)
-	clonedTags[azure.TagKeyAzdDeploymentTemplateHashName] = &templateHash
-
-	stackParams := map[string]*armdeploymentstacks.DeploymentParameter{}
-	for k, v := range parameters {
-		stackParams[k] = &armdeploymentstacks.DeploymentParameter{
-			Value: v.Value,
-		}
-	}
-
-	deploymentStackOptions, err := parseDeploymentStackOptions(options)
+	client, stack, err := d.validateOrDeployToSubscriptionHelper(ctx, subscriptionId, location, armTemplate, parameters, tags, options)
```

Comment:
I'm wondering if you may think it's a good idea to separate it along the lines of:

```golang
client, err := d.createClient(ctx, subscriptionId)
if err != nil {
  return nil, err
}

stack, err := d.stackFromArm(ctx, subscriptionId, location, armTemplate, parameters, tags, options)
if err != nil {
  return nil, err
}
```